core/blockly.js

```diff
@@ -381,6 +381,23 @@ Blockly.prompt = function(message, defaultValue, callback, _opt_title,
  */
 Blockly.statusButtonCallback = function(id) {
   window.alert('status button was pressed for ' + id);
+  Blockly.updateStatusButton(id, Blockly.StatusButtonState.READY);
+};
+
+/**
+ * Update the visual state of a status button in an extension category header.
```

Comment:
Now that you're not passing in a callback key, registering callbacks, etc., this doesn't need to be in blockly.js.  You can probably move this to your new file, skipping the need to go through the main workspace->flyout->button list, and delete statusButtonCallback entirely.

Reply:
We're overriding statusButtonCallback in the GUI [here](https://github.com/LLK/scratch-gui/blob/7fc01216d959e63ad70952123162b57420d34daf/src/containers/blocks.jsx#L61
), so I think it makes sense to keep that in blockly.js,

We're not using the updateStatusButton function yet, but that will also need to get called from the GUI.

Reply:
Okay, that makes sense.

Next question: can you pass a workspace into `updateStatusButton` instead of using `getMainWorkspace`?  It's better form not to use that function in new code--it's a holdover from when you could only have one blockly workspace on a page.